### PR TITLE
PR for #6 - Build basic Vert.x server

### PR DESCRIPTION
**Pull request by chess-equality**
#6 

Just added empty routes for now, will use the pages (Overview, etc.) when I go to #3.

### CODE REVIEW

**Comment by BFergerson** [comment]
@chess-equality, this isn't complete. You should be able to go to `localhost:8080/overview` (etc) and see the same thing you saw when you went to `overview.html` (etc).

Looks like it will just output text.

File: build.gradle.kts [hunk 1]
```diff
@@ -12,7 +12,7 @@ subprojects {
     repositories {
         mavenCentral()
         maven(url = "https://jitpack.io") { name = "jitpack" }
-        maven(url = "http://dl.bintray.com/kotlin/ktor") { name = "ktor" }
+        // maven(url = "http://dl.bintray.com/kotlin/ktor") { name = "ktor" }
```
Comment (BFergerson) [inline build.gradle.kts:15]: You can just remove this

File: server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt [hunk 2]
```diff
@@ -1,40 +1,56 @@
...
+import io.vertx.ext.web.RoutingContext
+import io.vertx.kotlin.core.http.listenAwait
+import io.vertx.kotlin.coroutines.CoroutineVerticle
+import io.vertx.kotlin.coroutines.dispatcher
+import kotlinx.coroutines.launch
+
+class PortalServer : CoroutineVerticle() {
+
+    override suspend fun start() {
+
```
Comment (BFergerson) [inline server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt:14]: I prefer no spacing on the first line after method declarations

File: server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt [hunk 3]
```diff
@@ -1,40 +1,56 @@
...
+    private suspend fun getTraces(ctx: RoutingContext) {
+        ctx.respond("/traces route")
+    }
+
+    private suspend fun getConfiguration(ctx: RoutingContext) {
+        ctx.respond("/configuration route")
+    }
+
+    private fun RoutingContext.respond(respondBody: String = "", status: Int = 200) {
+        this.response()
```
Comment (BFergerson) [inline server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt:40]: Only use `this` when necessary

**Comment by BFergerson** [comment]
@chess-equality, the reason why I don't believe serving empty routes is ideal is that switching to `/overview` (etc) means updating links in the HTML as everything in the HTML will point to `overview.html` (etc). Also, you will need to serve static assets via (StaticHandler) just as you would need to do even if you were rending HTML via kotlinx.html.

Serving the static files via this simple Vert.x server makes further steps easier. It also allows you to test each step to make sure no functionality is ever broken during this transition.

**Comment by chess-equality** [comment]
@BFergerson Thanks for all the comments, I will look at them tomorrow.

**Comment by chess-equality** [comment]
@BFergerson For this issue, do the routes become `localhost:8080/overview` (etc.) right away, or still `localhost:8080/tabs/overview.html` (etc.) ?

**Comment by BFergerson** [comment]
For this task please update the links so they no longer point to the `.html` files. Make sure you do a search for `html` and update it everywhere (even in the JS files).

**Comment by chess-equality** [comment]
Ok thanks.

**Comment by BFergerson** [comment]
@chess-equality, change the rendering to:
```kotlin
fun renderPage(): String {
        return buildString {
            appendLine("<!DOCTYPE html>")
            appendHTML().html {
```

Without `<!DOCTYPE html>` the UI looks like this (at least in Firefox):
image

**Comment by BFergerson** [comment]
@chess-equality, I'm still be redirected to `.html` pages. E.g. `href="traces.html"`

File: server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt [hunk 4]
```diff
@@ -1,40 +1,86 @@
...
+    override suspend fun start() {
+        // Build Vert.x Web router
+        val router = Router.router(vertx)
+        router.route().handler(LoggerHandler.create())
+        router.route().handler(ResponseTimeHandler.create())
+
+        // Static handler
+        router.route("/*").handler(StaticHandler.create())
+        // Routes
+        router.get("/overview*").coroutineHandler { ctx -> getOverview(ctx) }
```
Comment (BFergerson) [inline server/src/main/kotlin/com/sourceplusplus/portal/server/PortalServer.kt:40]: Remove "*" from these routes. Future versions may have sub pages. You'll need to remove all the `.html` links so it works correctly.

**Comment by BFergerson** [comment]
@chess-equality, delete copied resources from UI.

**Comment by chess-equality** [comment]
@BFergerson Please check latest commit.

**Pull request merged**